### PR TITLE
Add unit tests for ibc profiles feature

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR is adding tests for ibc profiles feature related #192.

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [ ] Updated the documentation. 
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: x/profiles/keeper/keeper_chain_link_test.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+				types.NewBech32Address(srcAddr, "cosmos"),
+				types.NewProof(srcPubKey, srcSigHex, srcAddr),
+				types.NewChainConfig("cosmos"),
+				suite.testData.profile.CreationDate,
+			),
+			shouldErr: true,
+			expStored: []sdk.AccAddress{},
+		},
+		{
+			name:  "user has no profile exists returns error",
```
Comment (bragaz) [inline x/profiles/keeper/keeper_chain_link_test.go:111]:
```suggestion
			name:  "user with no existent profile returns error",
```

File: x/profiles/keeper/keeper_chain_link_test.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+				types.NewBech32Address(srcAddr, "cosmos"),
+				types.NewProof(srcPubKey, srcSigHex, srcAddr),
+				types.NewChainConfig("cosmos"),
+				suite.testData.profile.CreationDate,
+			),
+			shouldErr: true,
+			expStored: []sdk.AccAddress{profileAcc},
+		},
+		{
+			name:  "invalid user exists returns error",
```
Comment (bragaz) [inline x/profiles/keeper/keeper_chain_link_test.go:98]:
```suggestion
			name:  "invalid user returns error",
```

File: x/profiles/keeper/keeper_chain_link_test.go [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+				types.NewBech32Address(srcAddr, "cosmos"),
+				types.NewProof(srcPubKey, srcSigHex, srcAddr),
+				types.NewChainConfig("cosmos"),
+				suite.testData.profile.CreationDate,
+			),
+			shouldErr: true,
+			expStored: []sdk.AccAddress{},
+		},
+		{
+			name: "valid condition returns no error",
```
Comment (bragaz) [inline x/profiles/keeper/keeper_chain_link_test.go:124]:
```suggestion
			name: "valid conditions returns no error",
```

File: x/profiles/keeper/keeper_chain_link_test.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+	tests := []struct {
+		name        string
+		store       func()
+		chainName   string
+		address     string
+		shouldFound bool
+		expRes      string
+	}{
+		{
+			name:        "Non existent link returns nothing",
```
Comment (bragaz) [inline x/profiles/keeper/keeper_chain_link_test.go:166]:
```suggestion
			name:        "Non existent link returns anything",
```

File: x/profiles/keeper/keeper_chain_link_test.go [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+			store:       func() {},
+			chainName:   "cosmos",
+			address:     suite.testData.user,
+			shouldFound: false,
+			expRes:      "",
+		},
+		{
+			name: "existent link returns error",
```
Comment (bragaz) [inline x/profiles/keeper/keeper_chain_link_test.go:174]:
```suggestion
			name: "existent link returns no error",
```

File: x/profiles/keeper/keeper_chain_link_test.go [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+			name:      "owner has no profile returns error",
+			store:     func() {},
+			owner:     suite.testData.otherUser,
+			chainName: "cosmos",
+			address:   suite.testData.otherUser,
+			shouldErr: true,
+			expStored: []sdk.AccAddress{},
+		},
+		{
+			name: "target address have not linked to the profile returns error",
```
Comment (bragaz) [inline x/profiles/keeper/keeper_chain_link_test.go:240]:
```suggestion
			name: "target address has not been linked to the profile returns error",
```

File: x/profiles/keeper/msg_server_chain_link_test.go [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+	blockTime := suite.testData.profile.CreationDate
+
+	tests := []struct {
+		name      string
+		store     func()
+		msg       *types.MsgLinkChainAccount
+		shouldErr bool
+		expEvents sdk.Events
+	}{
+		{
```
Comment (bragaz) [inline x/profiles/keeper/msg_server_chain_link_test.go:49]: we should also add a test to check that the unpack fail

File: x/profiles/keeper/relay_ibc_test.go [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+						destSigHex,
+						"invalid",
+					),
+				)
+			},
+			store:   func() {},
+			expPass: false,
+		},
+		{
+			name: "Store chain link faied returns error",
```
Comment (bragaz) [inline x/profiles/keeper/relay_ibc_test.go:102]:
```suggestion
			name: "Failed to store chain link returns error",
```

File: x/profiles/types/chain_link_test.go [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+		},
+		{
+			name: "invalid time returns no error",
+			chainLink: types.NewChainLink(
+				types.NewBech32Address(addr, "cosmos"),
+				types.NewProof(pubKey, sigHex, plainText),
+				types.NewChainConfig("cosmos"),
+				time.Time{},
+			),
+			expError: fmt.Errorf("createion time cannot be zero"),
```
Comment (bragaz) [inline x/profiles/types/chain_link_test.go:339]:
```suggestion
			expError: fmt.Errorf("creation time cannot be zero"),
```

**Review by bragaz — APPROVED** [review]

**Review by RiccardoM — APPROVED** [review]
Time for the final PR into the `master` branch 🎉

**Pull request merged**